nodkz/mongodb-memory-server
# Fix/improveMongoMemoryServer

- remove double check
- remove "null" use "undefined"
- convert one if to optional-chaining
- remove unnecessary "else" path
- test that "stop" returns "true" if not running
- test that "getDbPath" returns the dbPath
- add "assertion"
- change "getPort" to be sync
- change "getDbPath" to be sync
- change "getDbName" to be sync
- (singleDB) remove call to "getConnectionString" and use ".create"
- (multipleDB) remove call to "getConnectionString" and use ".create"
- remove deprecated function "getConnectionString"
- change "getUri" to be sync
- change "getInstanceInfo" to return undefined
- remove call to "ensureInstance" inside "stop"
- remove option "autoStart"
- change that "create" now always calls ".start"
- remove test "should create an instance but not autostart"
- refactor "start" to be more readable
- merge "runningInstance" and "instanceInfoSync" into "instanceInfo"
- test that "ensureInstance" returns early if already running
- move test "getUri" with "otherDb" into MongoMemoryServer-test
- test that "getUri" returns generated otherDb
- (MongoMemoryServer-test) refactor file to be more readable
- add assertion for sanity to check if "instanceInfo.instance" is defined
- always reset "port" to "undefined" if instance failed and not just when "defined"
- killProcess: fix "SIGINT"-"SIGKILL" warn condition
- remove "await" from "getUriBase" call
- remove comment & change log
- remove "MongoInstanceDataT.childProcess"
- remove "StartupInstanceData.uri"
- shorten "getUri"
- rename function "assertionInstanceInfoSync" to "assertionInstanceInfo"
- change "instanceOpts" to be readonly and Readonly
- change "binaryOpts" to be readonly and Readonly
- change "spawnOpts" to be readonly and Readonly
- remove resetting "port" inside "start"
- test that "getUri" returns correct value if no parameter is provided
- start: remove first ".catch"
- test(MongoMemoryServer): change test names to represent the tests
- test(multipleDB): await connection.close
- test(multipleDB): move "beforeAll" and "afterAll" hooks inside describe
- test(multipleDB): move "db1" and "db2" inside only using test
- test(singleDB): move "beforeAll" and "afterAll" hooks inside describe
- test(singleDB): await connection.close & move "db" inside test
- stop: reset "servers" after stopping them
- remove "stop" call where "start" is mocked
- change "MongoMemoryServer" to extend "EventEmitter"
- add enum "MongoMemoryServerStateEnum"
- add enum "MongoMemoryServerEventEnum"
- add function "stateChange"
- ensureInstance: fix starting calling "start" even when "start" was already called
- test that "ensureInstance" throws if "instanceInfo" is "undefined" but "_state" is "running"
- test that "ensureInstance" throws if "_state" has no case
- test that "ensureInstance" throws an error if _state is "starting" but emitting not "running"
- test that "ensureInstance" calls "start" and starts normally
- change "instanceInfo" to be "protected"
- rename "instanceInfo" to "_instanceInfo"
- rename function "getInstanceInfo" to "instanceInfo"
- change "instanceInfo" to be an getter instead of an function
- change "_state" to be "protected"
- add getter function "state" to get the current state (return "_state")
- remove function "getPort" in favor of ".instanceInfo.port"
- remove function "getDbPath" in favor of ".instanceInfo.dbPath"
- remove function "getDbName" in favor of ".instanceInfo.dbName"
